tests/dispatch/tests.py

```diff
@@ -53,22 +55,43 @@ def assertTestIsClean(self, signal):
         self.assertFalse(signal.has_listeners())
         self.assertEqual(signal.receivers, [])
 
-    def test_exact(self):
+    @override_settings(DEBUG=True)
+    def test_cannot_connect_no_kwargs(self):
+        def receiver_no_kwargs(sender):
+            pass
+        with self.assertRaisesRegex(ValueError, r'must accept keyword arguments'):
+            a_signal.connect(receiver_no_kwargs)
+        self.assertTestIsClean(a_signal)
+
+    def test_send(self):
         a_signal.connect(receiver_1_arg, sender=self)
         expected = [(receiver_1_arg, "test")]
         result = a_signal.send(sender=self, val="test")
         self.assertEqual(result, expected)
         a_signal.disconnect(receiver_1_arg, sender=self)
         self.assertTestIsClean(a_signal)
 
-    def test_ignored_sender(self):
+    def test_send_no_receivers(self):
+        result = a_signal.send(sender=self, val="test")
+        expected = []
```

Comment:
If I were writing these tests from scratch, I wouldn't use a separate `expected` variable everywhere (this is related to our preference for longer lines rather than a historical more strict adherence to 79 chars, I think).